deno/lib/stage2.ts

```diff
@@ -14,16 +14,28 @@ const getFunctionReference = (basePath: string, func: InputFunction, index: numb
   return {
     exportLine,
     importLine: `import ${importName} from "${url}";`,
+    metadata: {
+      url,
+    },
+    name: func.name,
   }
 }
 
-const getStage2Entry = (basePath: string, functions: InputFunction[]) => {
+export const getStage2Entry = (basePath: string, functions: InputFunction[]) => {
   const lines = functions.map((func, index) => getFunctionReference(basePath, func, index))
   const importLines = lines.map(({ importLine }) => importLine).join('\n')
   const exportLines = lines.map(({ exportLine }) => exportLine).join(', ')
-  const exportDeclaration = `export const functions = {${exportLines}};`
-
-  return [importLines, exportDeclaration].join('\n\n')
+  const metadata = lines.reduce(
+    (acc, { metadata, name }) => ({
+      ...acc,
+      [name]: metadata,
+    }),
+    {},
+  )
+  const functionsExport = `export const functions = {${exportLines}};`
```

Comment:
could you do the same here if you're already on it?

Reply:
This is not serialisable to JSON, though. The output of this will be something like:

```ts
export const functions = {"func1": func0, "func2": func1};
```

... where `func0` and `func1` are the imported modules.